common/stacks/src/main/scala/weco/api/stacks/services/SierraService.scala

```diff
@@ -104,16 +104,11 @@ class SierraService(
       //
       case Left(SierraErrorCode(132, specificCode, 500, _, _))
           if specificCode == 2 || specificCode == 929 =>
-        getStacksUserHolds(patron).flatMap {
-          case Right(holds)
-              if holds.holds
-                .map(_.sourceIdentifier)
-                .contains(sourceIdentifier) =>
-            Future.successful(Right(HoldAccepted.HoldAlreadyExists))
-
-          case Right(holds) if holds.holds.size >= holdLimit =>
+        getHolds(patron).flatMap {
+          case holds if holds.size >= holdLimit =>
             Future.successful(Left(HoldRejected.UserIsAtHoldLimit))
-
+          case holds if holds.keys.toList.contains(sourceIdentifier) =>
+            Future.successful(Right(HoldAccepted.HoldAlreadyExists))
           case _ => checkIfItemCanBeRequested(item)
         }
```

Comment:
I think you've introduced a bug here.

Consider: what if the user has a hold on this item *and* the user is at their hold limit?

Reply:
👍  re-ordering the case statements will fix that I think. I don't think we have a test for that particular case.